File: frontend/src/types/account.ts
// These are the frontend types for account data.
// We keep them separate so components and API files can reuse them.

export interface Account {
  id: number;
  name: string;
  industry: string | null;
  company_size: string | null;
  status: string;
  health_status: string | null;
  renewal_date: string | null;
  owner_user_id: number | null;
}

export interface AccountCreateRequest {
  name: string;
  industry?: string | null;
  company_size?: string | null;
  status?: string;
  health_status?: string | null;
  renewal_date?: string | null;
  owner_user_id?: number | null;
}

export interface AccountUpdateRequest {
  name?: string;
  industry?: string | null;
  company_size?: string | null;
  status?: string;
  health_status?: string | null;
  renewal_date?: string | null;
  owner_user_id?: number | null;
}

export interface ApiSuccessResponse<T> {
  success: true;
  message: string;
  data: T;
}

File: backend/app/schemas/account.py
# These schemas define the input and output structure
# for the accounts API.

from datetime import date

from pydantic import BaseModel


class AccountBase(BaseModel):
    """
    Shared account fields.
    """
    name: str
    industry: str | None = None
    company_size: str | None = None
    status: str = "active"
    health_status: str | None = None
    renewal_date: date | None = None
    owner_user_id: int | None = None


class AccountCreateRequest(AccountBase):
    """
    Request body for creating a new account.
    """
    pass


class AccountUpdateRequest(BaseModel):
    """
    Request body for updating an account.
    All fields are optional for partial updates.
    """
    name: str | None = None
    industry: str | None = None
    company_size: str | None = None
    status: str | None = None
    health_status: str | None = None
    renewal_date: date | None = None
    owner_user_id: int | None = None


class AccountResponse(BaseModel):
    """
    Response model for returning account data.
    """
    id: int
    name: str
    industry: str | None
    company_size: str | None
    status: str
    health_status: str | None
    renewal_date: date | None
    owner_user_id: int | None

    model_config = {"from_attributes": True}